sktime/transformations/panel/dictionary_based/_sfa.py

```diff
@@ -233,25 +268,44 @@ def transform(self, X, y=None, supplied_dft=None):
 
         return bags
 
-    def _transform_case(self, X, i, supplied_dft):
+    def _transform_case(self, X, supplied_dft=None):
         if supplied_dft is None:
```

Comment:
I am not sure, if I understand correctly, but I suppose you either choose between MFT and DFT? Wouldn't it be better to let the code decide, if the window_size too small compared to the word_length to fall back to DFT?

Reply:
supplied_dft is a multivariate TDE thing, I use the DFT from breakpoint creation to determine if a dimension is worth keeping or not and dont want to redo the processing. I kept it generally in transform because there may be some other use cases.

The DFT option is just for the initial DFT/FFT in the MFT and breakpoints, MFT is still used in the transform for both. I will edit it so the it switches between DFT/FFT depending on the word_length and window_size as suggested.

Reply:
I see :) 

just one minor suggestion: Maybe it would be more consistent with the sktime / sklearn to have a parameter in the constructor like `keep_dft=True` and store the dft within the the `fit` method if `keep_dft=True` and reuse it in the `transform` method?

Reply:
I am confused :) there is this switch alread:

```
        if self.save_binning_dft:
            self.binning_dft = dft
```

So, this is still not, what you are actually intending to do? :)

Reply:
Yeah, there is a parameter for saving the binning DFT in `fit`, but the `transform` part is kept general so it can be used with any DFT. 

The only usage of that is mine however. I dont mind changing it so that if you save the binning DFT, it is assumed you are going to use it in `transform` and removing the current `transform` parameter.

Reply:
Sorry for my stupid question: is there any other DFT (other than the MFT for sliding windows)?

Reply:
No, the option essentailly just replaces the MFT stage of the transform with the set of pre-calculated DFT's (in my case the disjoint windows from fit).

Reply:
In that case I would opt for using the `if self.save_binning_dft:` code. But I will not fight for it ;)

Reply:
Will go with that then, I kept it general for a potential use case which probably doesn’t exist.